reflred/footprint.py

```diff
@@ -235,15 +244,31 @@ def _abinitio_footprint(slit1, slit2, theta, width, offset=0.):
     p_near = (-width / 2 + offset) * sin(theta)
     p_far = (+width / 2 + offset) * sin(theta)
 
-    # beam profile is a trapezoid, 0 where a neutron entering at -s1/2 just
-    # misses s2/2, and 1 where a neutron entering at s1/2 just misses s2/2.
-    # With tiny front slits, the projection h2 may land on the other side
-    # of the beam center.  In this case the overall instensity is lower, but
-    # the trapezoid happens to have the same shape, with the flat region
-    # extending from -abs(w2) to abs(w2).  Since we only care about intensity
-    # relative to the beam for footprint correction, the scale factor cancels.
-    w1 = abs(d1/(d1-d2))*(s1x + s2x) / 2 - s1x / 2
-    w2 = abs(-abs(d1/(d1-d2)) * (s1x - s2x) / 2 + s1x / 2)
+    # source divergence sets a limit on all angles in the problem:
+    source_divergence = 180.0 if source_divergence is None else abs(source_divergence)
+    source_max_slope = np.tan(np.radians(source_divergence/2))
```

Comment:
Set `source_max_slope` directly if `source_divergence` is None.

I still prefer the consistency of storing divergence as 1-σ internally even if the calculation is done on FWHM.

Reply:
If we use 1-sigma width for the divergence, I have to think about what we would use here.  Would setting the "max angle" to the 1-sigma angle be appropriate?

Reply:
You can switch back to FWHM before doing the calculation, under the assumption that it is approximately uniform.

I settled on 1-σ as an internal format after fighting inconsistency bugs over the years when mixing representations. In particular, the `apply_abinitio_footprint` function pulls its default value from the refldata.source_divergence which should be stored as 1-σ.

I suppose `apply_abinitio_footprint` could convert from 1-σ `refldata.source_divergence` to FWHM so `_abinitio_footprint` can take FWHM if you feel that is cleaner. Please tag the parameter name with `_fwhm` if you do this as a form of documentation.